### DISCUSSION TITLE
Unable to start session when virtual sink ID is not found

### DISCUSSION THREAD

**Discussion by CrendKing** [LizardByte/community#215] [Sunshine]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the nightly release?

- [X] This issue is present in the nightly release

### Describe the Bug

I use the virtual sink to mute host during stream. I find the device (it is a "Realtek Digital Output") ID via tools/audio-info.exe. However, whenever I update the drive of this, the ID changes. The next time I stream, Sunshine will refuse to start session and report error ["Couldn't acquire Wave Format"](https://github.com/LizardByte/Sunshine/blob/d70d084f9fbb4e0150977a89d94937418a3ccf9c/tools/audio.cpp#L136) in the log. I have to physically go to the host computer to change that ID, which is inconvenient.

### Expected Behavior

I suggest instead of taking the device ID in the config, take a device name (e.g. "Realtek Digital Output"), which is unlikely to ever change. If multiple devices with same name exists, use the first one.

In any case, if a device is not found (either through the current ID method or the suggested name method), instead of report an error and stop the session, treat as no virtual sink is configured and start session normally. Ideally post a notification at bottom left telling user that virtual sink is not found.

### Host Operating System

Windows

### Operating System Version

Windows 11 22H2

### Architecture

64 bit

### Sunshine commit or version

0.19.1.191b929ab0f4de5c67a07896c88a9d4ec5de422a

### Package

Windows - portable

### GPU Type

Nvidia

### GPU Model

GeForce RTX 2070 Super

### GPU Driver/Mesa Version

531.41

### Config

```Shell
virtual_sink = {0.0.0.00000000}.{19373ed8-e943-4415-994a-e27c83f9c6af}
```

### Apps

```json
{
    "env": {
        "PATH": "$(PATH)"
    },
    "apps": [
        {
            "name": "Desktop",
            "image-path": "desktop.png"
        }
    ]
}
```

### Relevant log output

```shell
Everything else is normal. The only error is the one mentioned above, "Couldn't acquire Wave Format"
```

**Comment by Nonary** [comment]
Transferring this as a feature request, there is already information posted in log file about virtual sink not being found. Inserting notifications that display on the screen, or changing the audio sink to use names, are inherently new features, not a bug fix.

**Reply by CrendKing** [reply]
I agree mostly this is a feature request. However, in case owners of the project reject these requests, please at least make it ignore the wrong ID. Thanks!
